### ISSUE TITLE
Disable page buttons

### ISSUE THREAD

**Issue by Turnipisum** [bitfocus/companion#66]
**Describe the feature**
Option to disable the left 3 page buttons and use them for normal buttons.

**Is this platform dependent (windows, mac, ..)?**
no

**If documentation is required to implement, do you know where to find it?**
If applicable, add screenshots, protocol description PDFs, etc to help the developers along

**Usecases**
This would be great for when using more than one stream deck.

**Comment by dnmeid** [comment]
According to that I want to bump my suggestion about the configurable window-system.
Here is my writeup from different issue again:

After rethinking all of that I think we should drop the concept of the pages like they are at the moment and replace it with a more flexible solution. Also thinking of upcoming multi-device functionality. My proposal: Introduction of windows or whatever you want to call it. Basically a window should be a 2d array of buttons mapped onto physical keys. Like in a computer window, when the key area is smaller than the button array, overflow with scrolling can happen. Properties of a window are id or name, position on keys (top left), border size (w,h), pane size (w,h), pane position (x,y).
We should introduce companion as a new pseudo-device and offer actions to control it: window open at position (x/y and size w/h) (also brings window on top), window close, window scroll (direction and increment), window scroll to (x/y).
One can rebuild the workflow like it is with two windows. Window one occupying the three left buttons and window two occupying the remaining buttons. In window one you have buttons for scrolling window two up or down (or left and right).

The big advantage is, that you can have much more flexibility. Lets say I have only two pages. I can make one window with a button in top left "scroll to where page two is" and on page two in the same location one button with "scroll to where page one is".
This also is a solution for the sticky button request. Just make a second window and don't scroll it. If you only wan't to have one sticky button and it should be in the middle of the keys, no problem, just make sure window two is on top of window one. That means windows should not be brought to front when using a button.
Also the Page navigation system suggested by Deanbau could be user programmable that way, you just have to open a window with several buttons to new windows with several buttons to scroll positions. Ok it will take some time to program this as a user, but the point is that the power and flexibility is at the user end and they can customize their control surface the way they want.

Button naming could be w.y.x (w= window no, y= row no, x=column no). My suggestion is that the pane size is virtually infinite (fixed to some technical limit), so you can just place your buttons anywhere you want an scroll there, you don't have to grow the pane when you need more space and, more important, we don't need to reflow buttons when the pane shrinks.

Of course we need parameters for actions for that, but we need parameters anyway.

One more addition to the button flexibility (maybe I should do new issue): We should have a button pool. That means when creating a button with actions and so on it gets created in a pool and you can use the pool objects and place them in windows. With that you can reuse the button/have more instances of the same button. When editing the button you edit the pool object and all other instances reflect the edit. Copying a button in a window creates a new instance, copying in the pool creates an independent new button. We can also enumerate the pool objects and maybe put that number in the top right of a button.

**Comment by haakonnessjoen** [comment]
Leaning more to the idea from @Turnipisum, I was thinking of making Companion remember which pages devices are on.

Adding to this idea could be that you could change the page from admin, in the devices list. And thus we could add the posibility to optionally remove the page up/down buttons, from the upcoming config section.

That way, each device could use all 15 buttons, and you can configure them from admin. Or maybe even reassign pages from the devices using some "hold down x buttons for x seconds" action.

**Comment by dnmeid** [comment]
@willosof with my proposal all of this could be implemented by the user, but on the other hand all of this would have to be implemented by the user.
In the meantime I think the whole button system also could be provided by a module. That would mean the system like it is now with the easy to go layout could be a "easy buttons module" and I can provide the "fancy buttons module" later.

**Comment by willosof** [comment]
Well, what about the admin gui then?

**Comment by Fishymobile** [comment]
Another option would be implementing user defined buttons to move to different pages. Kind of like the profiles built into the elgato software. Could be a "navigation" module that removes the default page buttons and has page up, page down, and goto page # actions.

**Comment by IanTGreenwood** [comment]
At the very basic level, there could be an option in user prefs for "single page" mode.  It would be handy for when you only need 12 buttons, or you want to give an SD to a show caller or graphics op for some basic functions and don't want them to get confused by accidentally hitting the page button.

**Comment by willosof** [comment]
this is being worked on in the nopagebuttons branch :) fyi

**Comment by charlyschulte** [comment]
will the nopagebuttons branch be merged to the master ? and when this is planned to happen ?  
with friendly greetings
Charly Schulte

**Comment by willosof** [comment]
It will. It just needs more work before we can merge it, and we're swamped with other work these days :(

**Comment by josephdadams** [comment]
Closing because this is fixed in 2.0 - we don't need any more issue entries to track this very desirable feature :)

**Comment by matt-cook** [comment]
Is it fixed in the current 2.0 builds? I'm not seeing a no page option in the most recent and the branch looks pretty old now.

**Comment by istnv** [comment]
> Is it fixed in the current 2.0 builds? I'm not seeing a no page option in the most recent and the branch looks pretty old now.
In 2.0 you can delete the automatic page buttons.
